debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -150,6 +151,26 @@ public Lsn getMaxLsn() throws SQLException {
         }, "Maximum LSN query must return exactly one value"));
     }
 
+    public MaxLsnResult getMaxLsnResult(String alternativeMaxTransactionalQuery) throws SQLException {
+        Lsn maxLsn = queryAndMap(GET_MAX_LSN, singleResultMapper(rs -> {
+            final Lsn ret = Lsn.valueOf(rs.getBytes(1));
+            LOGGER.trace("Current maximum lsn is {}", ret);
+            return ret;
+        }, "Maximum LSN query must return exactly one value"));
+
+        if (StringUtils.isEmpty(alternativeMaxTransactionalQuery)) {
```

Comment:
Seems this would only be the case if someone set the statement explicitly to ""; is this what you're expecting here?

Reply:
The original idea for the configuration was that if you put nothing, the current flow wouldn't change, but with this second go around where the query is defaulted, but still overridable, I'm not sure what exactly we want to do.

Maybe we could make this as a sort of feature flag and just have a boolean for determining whether or not we utilize the new lsn handling?

Reply:
I think the question is whether there's a realistic chance for other statements really being used. If not (which I think is the case tbh.) then yes hard-coding the statement and instead having a feature-toggle for enabling/disabling this optimization seems more reasonable. It could remain an internal (undocumented one) as is right now, defaulting to enable this. Then this could be used to disable the optimization should it prove troublesome in production.